bin/oneoff/nces_data/import_school_stats_by_years_2018_2019_ccd

```diff
@@ -0,0 +1,127 @@
+#!/usr/bin/env ruby
+
+require_relative '../../../dashboard/config/environment'
+
+CDO.log = Logger.new(STDOUT)
+
+SURVEY_YEAR = '2018-2019'.freeze
+
+VIRTUAL_SCHOOL_MAP = {
+  'Full Virtual' => 'Yes',
+  'Missing' => nil,
+  'Not reported' => nil,
+  'Not Virtual' => 'No',
+  'Supplemental Virtual' => 'No',
+  'Virtual with face to face options' => 'Yes'
+}.freeze
+
+TITLE_I_MAP = {
+  'Title I targeted assistance eligible school-No program' => '1',
+  'Title I targeted assistance school' => '2',
+  'Title I schoolwide eligible-Title I targeted assistance program' => '3',
+  'Title I schoolwide eligible school-No program' => '4',
+  'Title I schoolwide school' => '5',
+  'Not a Title I school' => '6',
+  'Missing' => nil,
+  'Not reported' => nil
+}.freeze
+
+COMMUNITY_TYPE_MAP = {
+  '11' => 'city_large',
+  '12' => 'city_midsize',
+  '13' => 'city_small',
+  '21' => 'suburban_large',
+  '22' => 'suburban_midsize',
+  '23' => 'suburban_small',
+  '31' => 'town_fringe',
+  '32' => 'town_distant',
+  '33' => 'town_remote',
+  '41' => 'rural_fringe',
+  '42' => 'rural_distant',
+  '43' => 'rural_remote'
+}.freeze
+
+# @param unsanitized [String, nil] the unsanitized string
+# @returns [String, nil] the sanitized version of the string, with equal signs and double
+#   quotations removed. Returns nil on nil input, or if value is a dash (signifies missing in NCES data).
+def sanitize_string_for_db(unsanitized)
+  unsanitized = unsanitized == '-' ? nil : unsanitized
+  unsanitized&.tr('="', '')
+end
+
+AWS::S3.process_file('cdo-nces', "#{SURVEY_YEAR}/ccd/ccd_sch_029_1819_w_1a_091019.csv") do |filename|
+  SchoolStatsByYear.transaction do
+    SchoolStatsByYear.merge_from_csv(filename, {col_sep: ",", headers: true, quote_char: "\x00", encoding: 'ISO-8859-1:UTF-8'}) do |row|
+      {
+        school_id:          row['NCESSCH'].to_i.to_s,
```

Comment:
Why do we convert to integer and then convert back to string?

Reply:
Ahh great question. At one point, our schools table was only public and charter schools, which have IDs that are all numbers, but have leading zeros. They were imported and coerced to integers, so "012" became 12. Then, we added private schools, which have letters in the IDs (at which point, we moved the ID field in the schools table back to a string). So, this hack turns that "012" into 12, then back to "12" 🤦 .